src/Http/ResponseEmitter.php

```diff
@@ -202,15 +202,39 @@ protected function emitCookies(array $cookies)
     {
         foreach ($cookies as $cookie) {
             if (is_array($cookie)) {
-                setcookie(
-                    $cookie['name'],
-                    $cookie['value'],
-                    $cookie['expire'],
-                    $cookie['path'],
-                    $cookie['domain'],
-                    $cookie['secure'],
-                    $cookie['httpOnly']
-                );
+                if (version_compare(PHP_VERSION, '7.3.0', '>=')) {
+                    // PHP 7.3 and up don't support adding SameSite to the path,
+                    // but require the alternative setcookie syntax.
+                    // So we extract sameSite from the path
+                    $sameSite = null;
+                    $path = $cookie['path'];
+                    if (preg_match('/^(.*); SameSite=(.*)$/', $path, $matches) === 1) {
+                        $path = $matches[1];
+                        $sameSite = $matches[2];
+                    }
+                    setcookie(
+                        $cookie['name'],
+                        $cookie['value'],
+                        [
+                            'expires' => $cookie['expire'],
+                            'path' => $path,
+                            'domain' => $cookie['domain'],
+                            'secure' => $cookie['secure'],
+                            'httponly' => $cookie['httpOnly'],
+                            'samesite' => $sameSite,
+                        ],
```

Comment:
```suggestion
                        ]
```

This causes syntax error on PHP < 7.4

Reply:
Woops my bad